service/matching/matching_engine_test.go

```diff
@@ -2503,6 +2506,221 @@ func (s *matchingEngineSuite) TestDemotedMatch() {
 	task.finish(nil)
 }
 
+func TaskQueueMetricValidator(T *testing.T, capture *metricstest.Capture, FamilyCounterLength int, FamilyCounter float64, QueueCounterLength int, QueueCounter float64, QueuePartitionCounterLength int, QueuePartitionCounter float64) {
```

Comment:
just make all these methods on `matchingEngineSuite` so that you can use `s.Equal`, and don't have to pass around the T